tests/test_variable_visitor.py

```diff
@@ -22,66 +22,74 @@
+        visitor = VariableVisitor()
         tree = RealConstant('c0')
-        variables = tree.accept(VariableVisitor())
-        self.assertEqual(variables, {})
+        self.assertEqual({}, tree.accept(visitor))
+        tree = RealVariable('x')
+        self.assertEqual({'x': 'Real'}, tree.accept(visitor))
         tree = RealEquality(RealVariable('x'), RealVariable('y'))
-        variables = tree.accept(VariableVisitor())
-        self.assertEqual(variables, {'x': 'Real', 'y': 'Real'})
+        self.assertEqual({'x': 'Real', 'y': 'Real'}, tree.accept(visitor))
+
+    def test_string_visitor_easy(self):
+        visitor = VariableVisitor()
+        tree = StringLiteral('c0')
+        self.assertEqual({}, tree.accept(visitor))
+        tree = StringVariable('x')
+        self.assertEqual({'x': 'String'}, tree.accept(visitor))
+        tree = StringEquality(StringVariable('x'), StringVariable('y'))
+        self.assertEqual({'x': 'String', 'y': 'String'}, tree.accept(visitor))
 
     def test_boolean_visitor_inequality(self):
-        tree = BooleanEquality(BooleanVariable('x'), BooleanVariable('y'))
-        variables = tree.accept(VariableVisitor())
-        self.assertNotEqual(variables, {'x': 'Boolean', 'y': 'Bool', 'z': 'Bool'})
+        visitor = VariableVisitor()
+        tree_1 = BooleanEquality(BooleanVariable('x'), BooleanVariable('y'))
+        tree_2 = BooleanEquality(BooleanVariable('y'), BooleanVariable('z'))
+        self.assertNotEqual(tree_1.accept(visitor), tree_2.accept(visitor))
 
     def test_integer_visitor_inequality(self):
-        tree = IntegerEquality(IntegerVariable('x'), IntegerVariable('y'))
-        variables = tree.accept(VariableVisitor())
-        self.assertNotEqual(variables, {'x': 'Integer', 'y': 'Int', 'z': 'Int'})
+        visitor = VariableVisitor()
+        tree_1 = IntegerEquality(IntegerVariable('x'), IntegerVariable('y'))
+        tree_2 = IntegerEquality(IntegerVariable('y'), IntegerVariable('z'))
+        self.assertNotEqual(tree_1.accept(visitor), tree_2.accept(visitor))
 
     def test_real_visitor_inequality(self):
-        tree = RealEquality(RealVariable('x'), RealVariable('y'))
-        variables = tree.accept(VariableVisitor())
-        self.assertNotEqual(variables, {'x': 'Rational', 'y': 'Real', 'z': 'Real'})
+        visitor = VariableVisitor()
+        tree_1 = RealEquality(RealVariable('x'), RealVariable('y'))
+        tree_2 = RealEquality(RealVariable('y'), RealVariable('z'))
+        self.assertNotEqual(tree_1.accept(visitor), tree_2.accept(visitor))
+
+    def test_string_visitor_inequality(self):
+        visitor = VariableVisitor()
+        tree_1 = StringEquality(StringVariable('x'), StringVariable('y'))
+        tree_2 = StringEquality(StringVariable('y'), StringVariable('z'))
+        self.assertNotEqual(tree_1.accept(visitor), tree_2.accept(visitor))
 
     def test_boolean_visitor_hard(self):
         tree = BooleanEquality(
```

Comment:
Here and later in the file the parenthesis style is also different

Reply:
Yep, I'll stick to the more compact version. I also noticed we're inconsistent with single and double quotation marks, but that's more effort to fix.

Reply:
Yep, we should ideally have a linter/formatter at some point, so that the style is automatically enforced